Sprint #19.adoc

```diff
@@ -29,9 +29,45 @@
 * TASK: make hasAttachment searchable via mailbox-api
 * TASK: make GetMessagesList call the search api
 
+Remark. If you want to call here the search API for Messages you will have to problems :
+ - It returns a list of UID. You may need to add information about mailboxes depending on your implementations choices
+ - As such the returned document is not present (reduce network load) + the index document is not the original mail (text extraction). So you will still need a read to Cassandra after your read to ElasticSearch..
+
 == Questions
 
 * Do we want to manage quota on upload ? Ask that with INBOX and UX team
+
+From the spec : 
+
+....
+If uploading a file would take the user over quota, the server SHOULD delete previously uploaded (but unused) files before their expiry time. 
+....
+
+I guess so.
+
+Beware thow. IMAP quota do not have any concept like "user". It uses QuotaRoots. Mailboxes is attached to a QuotaRoot. The current implementation matches the notion of user, but you have no warranty quotaRoot definition will match the notion of user mailboxes.
+
 * How hard would it be to filter on mailbox and/or users when use Filter on GetMessagesList
+
+Several cases :
+
+ - No filters means all messages the user can see. Then we need a user fileds added while indexing. Note that supporting shared mailboxes here will be hard....
+ - Write consideration. We should ensure a user have the right to search a given mailbox. Note that current implementation do not check this.... If you use it directly with JMAP, where user can specify __any__ mailboxId in its request, this becomes a vulnerability.
+
 * How hard is it the handle mailbox rename for search ?
+
+Yes it is. All depends how you represent mailboxes. With the **SimpleMessageSearchIndex**, as it is built on top of MessageMapper fetching, you do not need anything.
+
+With ElasticSearch, it is more complicated as you need to reindex it, at least partially (mailbox field), or even completly with *parent-child* relationShpip.
```

Comment:
Use the ReIndexer upon mailboxRename then, it would be the less time consuming stuff to do IMO